source/common/common/utility.cc

```diff
@@ -486,6 +486,33 @@ std::string StringUtil::removeCharacters(const absl::string_view& str,
   return absl::StrJoin(pieces, "");
 }
 
+absl::optional<uint64_t> StringUtil::readAndRemoveLeadingDigits(absl::string_view& s) {
+  absl::optional<uint64_t> output;
+  const char* p = s.data();
+  const char* limit = p + s.size();
+  uint64_t val = 0;
+  while (p < limit) {
+    const char c = *p;
+    if (c < '0' || c > '9') {
+      break;
+    }
+    uint64_t new_val = (val * 10) + (c - '0');
+    if (new_val / 8 < val) {
```

Comment:
Why divide by 8?

Reply:
It's not an exact formula, as dividing by any number from 5 through 10 would work in this case to test for overflow. 8 is kinda convenient cause it's just `<< 3`, which makes it run a touch faster, so why not use that over the other options.